### PR TITLE
feat(evm): Deduct a fee during the creation of a FunToken mapping. Implemented by `deductCreateFunTokenFee` inside of the `eth.evm.v1.MsgCreateFunToken` transaction.

### PR DESCRIPTION
**Pull request by Unique-Divine**
# Purpose / Abstract

1. Completes a TODO comment inside the "eth.evm.v1.MsgCreateFunToken" transaction
2. Organizes the code in the "erc20.go" file more since the file was becoming quite large. Now, the functionality specific to FunToken creation is in "funtoken_from_erc20.go" and "funtoken_from_coin.go", while the ERC20 call functions live inside "erc20.go". 

Part of #1898 and #1836 

## Summary by CodeRabbit

- **New Features**
  - Introduced fee deduction during the creation of `FunToken` mappings in the Ethereum Virtual Machine (EVM).

- **Bug Fixes**
  - Ensured accounts are funded appropriately before operations in `TestCreateFunTokenFromERC20`, `TestCreateFunTokenFromCoin`, and `TestSendFunTokenToEvm`.

- **Refactor**
  - Replaced `MINTER_ROLE` role with an "owner" role in `ERC20Minter`.
  - Renamed function `OnStart` to `OnRunStart` for clarity.

- **Documentation**
  - Updated comments in proto files for better parameter clarity in contract creation and function calls.

### CODE REVIEW

File: x/evm/keeper/precompiles.go
Comment (Unique-Divine) [inline x/evm/keeper/precompiles.go:1]: Unused code here, so we delete it

File: x/evm/embeds/ERC20Minter.sol
Comment (Unique-Divine) [inline x/evm/embeds/ERC20Minter.sol:1]: Fixed some of the doc comments in this file, as some were not accurate

File: x/evm/errors.go
Comment (Unique-Divine) [inline x/evm/errors.go:1]: Defined by https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.9.6/contracts/access/Ownable.sol

File: x/evm/keeper/erc20.go [hunk 1]
```diff
@@ -272,6 +220,16 @@ func (k Keeper) CallContractWithInput(
...
+// computeCommitGasLimit: If the transition is meant to mutate state, this
+// function computes an appopriates gas limit for the call with "contractInput"
+// bytes against the given contract address.
+//
+// ℹ️ This creates a cached context for gas estimation, which is essential
+// because state transitions can occur outside of the EVM that are triggered
+// by Ethereum transactions, like in the case of precompiled contract or
+// custom EVM hook that runs after tx execution. Gas estimation in that case
+// could mutate the "ctx" object and result in falty resulting state, so we
+// must cache here to avoid this issue.
```
Comment (Unique-Divine) [inline x/evm/keeper/erc20.go:232]: Big explainer added here since the reasoning behind the `cachedCtx` is nuanced

File: x/evm/evmtest/erc20.go [hunk 2]
```diff
@@ -23,7 +24,7 @@ func DoEthTx(
 
 func AssertERC20BalanceEqual(
 	t *testing.T,
-	deps *TestDeps,
+	deps TestDeps,
 	contract, account gethcommon.Address,
 	balance *big.Int,
 ) {
```
Comment (Unique-Divine) [inline x/evm/evmtest/erc20.go:30]: This should take a non-pointer receiver because it's not modifying state and only reading it

**Review by k-yang — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -82,6 +82,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [#1959](https://github.com/NibiruChain/nibiru/pull/1959) - feat(evm): Add precompile to the EVM that enables trasnfers of ERC20 tokens to "nibi" accounts as regular Ethereum transactions 
 - [#1960](https://github.com/NibiruChain/nibiru/pull/1960) - test(network): graceful cleanup for more consistent CI runs
 - [#1961](https://github.com/NibiruChain/nibiru/pull/1961) - chore(test): reverted funtoken precompile test back to the isolated state
+<<<<<<< HEAD
```
Comment (Unique-Divine) [inline CHANGELOG.md:85]:
```suggestion
```

**Pull request merged**